### ISSUE TITLE
False Positive: "react/prop-types [PROP_NAME] is missing in props validation" While sending through parent

### ISSUE THREAD

**Issue by ExpectoPatrom** [yannickcr/eslint-plugin-react#2714]
In my App.js I'm sending an element of the state. Then I use it in the child that receive the data. It works, but the eslint server gives me that error.
 `react/prop-types [PROP_NAME] is missing in props validation`
Here's the section that sends the data:
`<Menu dishes={this.state.dishes} />`
And I use it in
```
const menu = this.props.dishes.map((dish) => {
      return ("some stuff")
```
As said above it works on the browser

**Comment by ljharb** [comment]
So, a few things: first, I'd need the entire code of the component. Second, though, whether it works in the browser is irrelevant; most linting rules aren't about "make the code work", they're about "make the code readable and maintainable and consistent".

In this case, the likely situation is that `Menu.propTypes` doesn't have `dishes` defined.

**Comment by ljharb** [comment]
Closing until there's enough code provided to reproduce.
